### PR TITLE
Allow using sinks outside of unsafe mode

### PR DESCRIPTION
**Pull request by bkirwi**
Marking sinks as out of unsafe mode: we've landed the most significant changes, and this needs to hit the next release window if we don't want to slow down onboarding.

### Motivation

On the checklist for #11503!

### Checklist
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - Support the CREATE SINK command outside unsafe mode.

### CODE REVIEW

**Comment by sploiselle** [comment]
@bkirwi Meta nit; need to amend the release note. From `doc/developer/guide-changes.md`:

> #### How to write a good release note
> 
> Every release note should be phrased in the imperative mood, like a Git
> commit message. They should complete the sentence, "This release will...".
> 
> Good release notes:
> 
>   - [This release will...] Require the `-w` / `--workers` command-line option.
>   - [This release will...] In the event of a crash, print the stack trace.
> 
> Misbehaved release notes:
> 
>   - Users must now specify the `-w` / `-threads` command line option.
>   - Materialize will print a stack trace if it crashes.
>   - Instead of limiting SQL statements to 8KiB, limit them to 1024KiB instead.

**Review by sploiselle — APPROVED** [review]

**Comment by benesch** [comment]
I think there are probably a few sink options we should continue to gate behind unsafe mode! I’m AFK now but will look later. I assume approximately everything besides SIZE and PROGRESS TOPIC should be unsafe-only options.

Speaking of, did the PROGRESS TOPIC work land already?

**Comment by bkirwi** [comment]
It looks like the only current option supported directly on sinks (ie. in the WITH block) is `SNAPSHOT`. `PROGRESS TOPIC` is a new connection-level option; `SIZE` isn't supported yet for sinks, though it should be. (And I'm not sure we're tracking that -- mentally noted.) But I'm not sure I know what I should be looking for here; happy to do as y'all suggest!

Progress topic is just in: https://github.com/MaterializeInc/materialize/pull/14526

**Comment by benesch** [comment]
> `SIZE` isn't supported yet for sinks, though it should be. (And I'm not sure we're tracking that -- mentally noted.)

I'm happy to report that it is! https://github.com/MaterializeInc/materialize/issues/13889

> It looks like the only current option supported directly on sinks (ie. in the WITH block) is `SNAPSHOT`. `PROGRESS TOPIC` is a new connection-level option...  But I'm not sure I know what I should be looking for here; happy to do as y'all suggest!

There's a bunch of other Kafka connection-specific options that are suspect! I wrote up my thoughts here: #14656. But the easiest thing to do is to just mark those all as unsafe. Just need a check similar to the one we have for sources:

https://github.com/MaterializeInc/materialize/blob/37b70aa057428aec96205cb002e71fd2b7ceb84a/src/sql/src/plan/statement/ddl.rs#L405-L412

At least, I don't _think_ we have such a check already. It's possible I wasn't looking closely enough.

**Comment by bkirwi** [comment]
I think this is the check you have in mind, yeah?

https://github.com/MaterializeInc/materialize/blob/c60780f4397ba6bc5ef1392dd18ab4a0fe8014a9/src/sql/src/plan/statement/ddl.rs#L2079-L2086

**Comment by benesch** [comment]
Oh beautiful, yes!

**Comment by bkirwi** [comment]
Cool! IIUC we're planning to wait until after this week's release to merge regardless, but otherwise it sounds like this is ready to go.

**Comment by benesch** [comment]
v0.27.0-alpha.19 has been cut. This looks safe to merge to me whenever now!

**Comment by benesch** [comment]
I think there might be some librdkafka options (i.e., options that go after
KAFKA CONNECTION), besides PROGRESS TOPIC, that ought to be unsafified if
they aren’t already.

**Pull request merged**
